fission/src/systems/scene/GizmoSceneObject.ts

```diff
@@ -0,0 +1,194 @@
+import * as THREE from "three"
+import SceneObject from "./SceneObject"
+import { TransformControls } from "three/examples/jsm/controls/TransformControls.js"
+import InputSystem from "../input/InputSystem"
+import World from "../World"
+import MirabufSceneObject from "@/mirabuf/MirabufSceneObject"
+import { Object3D, PerspectiveCamera } from "three"
+import { ThreeMatrix4_JoltMat44, ThreeQuaternion_JoltQuat } from "@/util/TypeConversions"
+import { RigidNodeReadOnly } from "@/mirabuf/MirabufParser"
+
+export type GizmoMode = "translate" | "rotate" | "scale"
+
+class GizmoSceneObject extends SceneObject {
+    private _gizmo: TransformControls
+    private _obj: Object3D
+    private _parentObject: MirabufSceneObject | undefined
```

Comment:
```suggestion
    private _parentObject?: MirabufSceneObject
```

I prefer this just because it's cleaner and is also what you do in the constructor.